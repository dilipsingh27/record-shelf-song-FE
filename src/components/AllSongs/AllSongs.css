.container {
    /* max-width:1000px; */
    background-color: #F5F5F5;
    margin: 0 150px 0 150px;
    
    
}

.container-title {
    height: 100px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: row;
}
.container-title p {
    font-weight: 700;
    font-size: 40px;
    
}

.container img {
    /* height:40px;
    width: 40px; */
}

.main-padding {
    padding: 0 120px 0 120px;
}

.all-cards {
    display:flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin:0 120px 0 120px;
}