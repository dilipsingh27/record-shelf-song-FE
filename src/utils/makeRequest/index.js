import axios from 'axios';
import { BACKEND_URL } from "../../constants/apiEndpoints";
import { ERROR_ROUTE } from '../../constants/routes';

const makeRequest = async (
    apiEndPoint,
    dynamicConfig = {},
    navigate
  ) => {
    try {
      // console.log(dynamicConfig)
    const requestDetails = {
      //   baseURL: BACKEND_URL,
        url: `${BACKEND_URL}${apiEndPoint.url}`,
        method: apiEndPoint.method,
        ...dynamicConfig,
      };
  
      const { data } = await axios(requestDetails);
      // const res  = await axios(requestDetails);
      // console.log(res)
      // console.log(res.data[0].claps)
      // console.log(res.data)
      // return res;
      return data;
    } catch (e) {
      console.log(e);
      
      if(navigate) {
        const errorStatus = e.response?.status
        if(errorStatus){
          navigate(`${ERROR_ROUTE}/${errorStatus}`)
        }
        else {
          navigate(ERROR_ROUTE)
        }
      }
    }
    
  };
  
  export default makeRequest;