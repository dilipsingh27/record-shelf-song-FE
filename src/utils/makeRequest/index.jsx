import axios from 'axios';
const BACKEND_URL = 'http://localhost:8080/';

const makeRequest = async (apiEndPoint, dynamicConfig, navigate) => {
    try {
        const requestDetails = {
            baseURL: BACKEND_URL,
            url: apiEndPoint.url,
            method: apiEndPoint.method,
            headers: {
                authorization: 'Bearer QWlzaHdhcnlhIE4=',
            },
            ...dynamicConfig,
        };
        const { data } = await axios(requestDetails);
        return data;
    } catch (error) {
        const errorCode = error.response?.status;
        if (errorCode) {
            navigate(`/error/${errorCode}`);
        } else {
            navigate('/error');
        }
    }
};
  
export default makeRequest;