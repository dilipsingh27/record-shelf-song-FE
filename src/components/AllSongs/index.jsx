import React, { useEffect,useState } from 'react';
import './AllSongs.css'
import iconGenre from '../../assets/icon-genre.svg'
import {Link} from 'react-router-dom'
import SongCard from '../SongCard';
import axios, { all } from 'axios';


const AllSongs = () => {

    const[allSong,setAllSong] = useState([]);
    const [error,setError] = useState();
    const [countLike,setCountLike] = useState([]);
    const [heart,setHeart] = useState([]);
    
    const handleData = (d)=> {
        setAllSong(d)
    }
    
    useEffect( ()=> {
        const getData = async () => {
            const songsFromServer = await fetchData()
            // console.log(songsFromServer);
            // setAllSong(songsFromServer.data)
            setAllSong(songsFromServer)
            // handleData(songsFromServer.data)
            // console.log(allSong)
        }
        getData()
    },[])
    
    const fetchData = async() => {
        try {
            const response = await fetch('http://localhost:8080/api/records',
            {headers:{Authorization:'Bearer QWlzaHdhcnlhIE4='}})
            // const response = await axios.get('http://localhost:8080/api/records',
            // {headers:{Authorization:'Bearer QWlzaHdhcnlhIE4='}})
            // console.log(response.json())
            const {data} =  await response.json()
            // setAllSong(data)
            // console.log(typeof(data));
            // console.log(data)
            // console.log("hgfhghjgj")
    
            // console.log(allSong)
            return data;
            
        } catch (error) {
            setError(error.message)
        }
    }
    
    useEffect(()=> {
        allSong.map(async(song) => {
        
                const {data} = await axios.get(`http://localhost:8080/api/records/${song.id}/likes`,
                {headers:{Authorization:'Bearer QWlzaHdhcnlhIE4='}})
                // console.log(data.data)
                const {count,like} = data.data;
                // console.log(count)
                // setCountLike([...this.countLike,count])
                
                
                // setCountLike(countLike.push(count))


                // const a = []
                // a.push(count)
                // setCountLike(a)
                // setCountLike(countLike.concat(count))
                
                // setCountLike(countLike => countLike.concat(count))
                setCountLike(countLike => [...countLike,count])
                setHeart(heart => [...heart,like])
                // console.log(countLike)
                
        })
        
    },[allSong])
    
    // console.log(countLike)
    // console.log(heart)
    
    // const fetchLike = async (songID) => {
    //     const res = await axios.get(`http://localhost:8080/api/records/${songID}/likes`,
    //     {headers:{Authorization:'Bearer QWlzaHdhcnlhIE4='}})
    //     return res.data
    // }


    // const likeData = allSong.map(song => {
    //     return (
    //         fetchLike(song.id)
    //     )
    // })
    // console.log(allSong)
    
    
    return (
       <section className='container'>
        <div className="container-title main-padding">
            <p>all Songs</p>
            <Link to="/genre">
            <img src={iconGenre} alt="efrrefe" />
            </Link>
        </div>
        
        
        <div className="all-cards">
            {
                allSong.map((song,index) => {
                   return ( <SongCard key={song.id} 
                                    id={song.id}
                                    name={song.name}
                                    imageUrl={song.imageUrl}
                                    artist={song.artist}
                                    genre={song.genre}
                                    like={countLike[index]}
                                    heart={heart[index]}
                            />
                        )
                })
            }
        </div>
       </section>
    )
}

export default AllSongs