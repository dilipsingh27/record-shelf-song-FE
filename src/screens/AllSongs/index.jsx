import React, { useEffect, useState } from 'react';
import './AllSongs.css';
import iconGenre from '../../assets/icon-genre.svg';
import { Link, useNavigate } from 'react-router-dom';
import SongCard from '../../components/SongCard';
import makeRequest from '../../utils/makeRequest';
import { GET_SONGS } from '../../constants/apiEndpoints';



const AllSongs = () => {

    const [allSong, setAllSong] = useState([]);
    // const [error,setError] = useState();
    const navigate = useNavigate();

    useEffect(() => {
        makeRequest(GET_SONGS, {}, navigate)
            .then((response) => {
                setAllSong(response.data);
            })
            .catch(() => {

            });
    }, []);





    return (
        <section className='container' data-testid = 'songs'>
            <div className="container-title main-padding">
                <p>all Songs</p>
                <Link to="/genre" data-testid = 'genre'>
                    <img src={iconGenre} alt="efrrefe" />
                </Link>
            </div>


            <div className="all-cards" >
                {
                    // allSong.map((song,index) => {
                    allSong.map((song) => {
                        return (<SongCard data-testid = 'Card'
                            key={song.id}
                            id={song.id}
                            name={song.name}
                            imageUrl={song.imageUrl}
                            artist={song.artist}
                            genre={song.genre}
                            
                        />
                        );
                    })
                }
            </div>
        </section>
    );
};

export default AllSongs;