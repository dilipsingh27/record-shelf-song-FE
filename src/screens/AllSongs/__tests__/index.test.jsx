import AllSongs from "..";
import { render, screen,waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import makeRequest from "../../../utils/makeRequest";

jest.mock("../../../utils/makeRequest");

const mockData = {
    data: [
        {
            id: "45e1d753-2986-43cb-9b9d-30c370056319",
            name: "This Love",
            genre: {
              id: "128aa7f8-c943-48ce-b352-7edd26fa4c6e",
              name: "Pop",
            },
            artist: {
              id: "496b0a85-2bfa-45bc-8d0f-57fe0ce55708",
              name: "Maroon 5",
            },
            imageUrl:
              "https://i.scdn.co/image/ab67616d0000b27317b3850d758fff5a2301e537",
            publishedAt: "2002-06-25T00:00:00",
        }, 
    ],
};

describe('Song Snapshot', () => {
    it('should match snapshot', () => {
        makeRequest.mockResolvedValue(mockData);
        const { asFragment } = render(
            <BrowserRouter>
                <AllSongs />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render all songs', async () => {
        makeRequest.mockResolvedValue(mockData);
        render(
            <BrowserRouter>
                <AllSongs />
            </BrowserRouter>
        );
        await waitFor(() => {
            const songCards = screen.getByText('all Songs')
            expect(songCards).toBeInTheDocument();
        })
    })

    // it('should render exact number of the cards', async () => {
    //     makeRequest.mockResolvedValue(mockData);
    //     render(
    //         <BrowserRouter>
    //             <AllSongs />
    //         </BrowserRouter>
    //     );
    //    await waitFor(() => {
    //     const songCards = screen.getAllByTestId('Card');
    //     expect(songCards.length).toEqual(1);
    //    })
    // })

    it('should render the container component without crashing when page is loaded', async () => {
        makeRequest.mockResolvedValue(mockData);
        render(
            <BrowserRouter>
                <AllSongs />
            </BrowserRouter>
        );
        await waitFor(() => {
            const container = screen.getByTestId('songs');
            expect(container).toBeInTheDocument();
        })
    })

    it('should navigate to genre page when genre icon is clicked', async () => {
        makeRequest.mockResolvedValue(mockData);
        render(
            <BrowserRouter>
                <AllSongs />
            </BrowserRouter>
        );
        await waitFor(() => {
            const genreIcon = screen.getByTestId('genre');
            expect(genreIcon).toBeInTheDocument();
        })
    })


})

