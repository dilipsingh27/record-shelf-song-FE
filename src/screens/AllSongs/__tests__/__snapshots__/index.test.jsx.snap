// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song Snapshot should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="songs"
  >
    <div
      class="container-title main-padding"
    >
      <p>
        all Songs
      </p>
      <a
        data-testid="genre"
        href="/genre"
      >
        <img
          alt="efrrefe"
          src="icon-genre.svg"
        />
      </a>
    </div>
    <div
      class="all-cards"
    />
  </section>
</DocumentFragment>
`;
